msmbuilder/featurizer/featurizer.py

```diff
@@ -513,37 +537,66 @@ def __init__(self, types=['phi', 'psi'], n_bins=18, kappa=20.):
                              (str(known), str(types)))
 
     def describe_features(self, traj):
-        """Return a list of dictionaries describing the von Mises features."""
-        x = []
-        for a in self.types:
-            func = getattr(md, 'compute_%s' % a)
-            aind, _ = func(traj)
-            n = len(aind)
-
-            resSeq = [(np.unique([traj.top.atom(j).residue.resSeq for j in i]))
-                      for i in aind]
-            resid = [(np.unique([traj.top.atom(j).residue.index for j in i]))
-                     for i in aind]
-            resnames = [[traj.topology.residue(j).name for j in i]
-                        for i in resid]
-
-            bigclass = ["dihedral"] * n * self.n_bins
-            smallclass = [a] * n * self.n_bins
-            aind = list(aind) * self.n_bins
-            resnames = resnames * self.n_bins
-            resSeq = resSeq * self.n_bins
-            resid = resid * self.n_bins
-            otherInfo = n * (self.n_bins*"vm%s," %
-                             tuple(range(self.n_bins))).split(',')[:-1]
-
-            for i in range(len(resnames)):
-                d_i = dict(resname=resnames[i], atomind=aind[i],
-                           resSeq=resSeq[i], resid=resid[i],
-                           otherInfo=otherInfo[i], bigclass=bigclass[i],
-                           smallclass=smallclass[i])
-                x.append(d_i)
+        """Return a list of dictionaries describing the dihderal features.
+
+        Parameters
+        ----------
+        traj : mdtraj.Trajectory
+            The trajectory to describe
+
+        Returns
+        -------
+        feature_descs : list of dict
+            Dictionary describing each feature with the following information
+            about the atoms participating in each dihedral
+                - resnames: unique names of residues
+                - atominds: the four atom indicies
+                - resseqs: unique residue sequence ids (not necessarily
+                  0-indexed)
+                - resids: unique residue ids (0-indexed)
+                - featurizer: Dihedral
+                - featuregroup: The bin index(0..nbins-1)
```

Comment:
Shouldn't "bin" be an additional field, or no? So one column is phi-psi and another column is bin. Whatever you think is best here for your use cases.